### PR TITLE
Fix the certman operator logic to look up HZ ID by DNSZone.Status.AWS.ZoneID in Hive

### PR DESCRIPTION
**Pull request by anispate**
[OSD-17701](https://issues.redhat.com/browse/OSD-17701) - Fix the certman operator logic to look up HZ ID by DNSZone.Status.AWS.ZoneID in Hive.

### CODE REVIEW

File: pkg/clients/aws/route53.go [hunk 1]
```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
...
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		return "", err
+	}
+
+	if len(dnsZones.Items) == 0 {
+		return "", fmt.Errorf("no dns zones found")
 	}
+	dnsZone := dnsZones.Items[0]
```
Comment (mjlshen) [inline pkg/clients/aws/route53.go:139]: Are there any situations where we have >1 dnszones (do we care)?

File: pkg/clients/aws/route53.go [hunk 2]
```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
 		return "", err
 	}
 
-	baseDomain := cr.Spec.ACMEDNSDomain
-	if !strings.HasSuffix(baseDomain, ".") {
-		baseDomain = baseDomain + "."
+	dnsZones := hivev1.DNSZoneList{}
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
```
Comment (cblecker) [inline pkg/clients/aws/route53.go:130]: As this is a new list call, does the operator have sufficient permissions to do this list?

File: pkg/clients/aws/route53.go [hunk 3]
```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
...
 	}
+	dnsZone := dnsZones.Items[0]
+
+	dnsZonePath := dnsZone.Status.AWS.ZoneID
+	dnsZoneId := filepath.Base(*dnsZonePath)
 
 	for _, hostedzone := range hostedZones {
-		if strings.EqualFold(baseDomain, *hostedzone.Name) {
+		if strings.EqualFold(dnsZoneId, *hostedzone.Id) {
 			zone, err := c.client.GetHostedZone(&route53.GetHostedZoneInput{Id: hostedzone.Id})
```
Comment (mjlshen) [inline pkg/clients/aws/route53.go:146]: Do we even need this loop/if anymore? We know exactly what the ID should be and can just `route53:GetHostedZone` with `dnszone.Status.AWS.ZoneID`?

File: pkg/clients/aws/route53.go [hunk 4]
```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
...
-		baseDomain = baseDomain + "."
+	dnsZones := hivev1.DNSZoneList{}
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		return "", err
+	}
+
+	if len(dnsZones.Items) == 0 {
+		return "", fmt.Errorf("no dns zones found")
```
Comment (cblecker) [inline pkg/clients/aws/route53.go:137]: Can this error be a bit more descriptive? If I was to get this error message I would be thinking the route53 zone was missing, but that's not what it's checking. It's checking if there are and DNSZone objects in a specific namespace.

File: pkg/clients/aws/route53.go (hunk 1 above)
Comment (cblecker) [inline pkg/clients/aws/route53.go:139]: This assumes you are only getting one zone back. Can there be multiple objects? Should we error if there is more than one?

File: pkg/clients/aws/route53.go [hunk 5]
```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
...
 	}
+	dnsZone := dnsZones.Items[0]
+
+	dnsZonePath := dnsZone.Status.AWS.ZoneID
+	dnsZoneId := filepath.Base(*dnsZonePath)
```
Comment (cblecker) [inline pkg/clients/aws/route53.go:142]:
```suggestion
	dnsZoneId := filepath.Base(*dnsZone.Status.AWS.ZoneID)
```

Does the following work to eliminate the need to create a new in-memory object for `dnsZonePath`?

File: pkg/clients/aws/route53_test.go [hunk 6]
```diff
@@ -261,8 +286,7 @@ func setUpEmptyTestClient(t *testing.T) (testClient client.Client) {
 	s.AddKnownTypes(certmanv1alpha1.GroupVersion, certRequest)
 
 	// aws is not an existing secret
-	objects := []runtime.Object{certRequest}
-
+	objects := []runtime.Object{certRequest, awsSecret, testClusterDeployment, testDnsZone}
```
Comment (mjlshen) [inline pkg/clients/aws/route53_test.go:289]:
```suggestion
	objects := []runtime.Object{certRequest}
```

**Review by mjlshen — APPROVED** [review]
/lgtm

**Pull request merged**